### ISSUE TITLE
References to a class in an imported module fails

### ISSUE THREAD

**Issue by pyjsorg** [pyjs/original-pyjs#633]
The following code doesn't work:

```
import ui

...

class MyPanel(ui.Panel):
    def __init__(self):
        ui.Panel.__init__(self)
        ...
```

It seems that any reference to a method of a class in an imported module
fails.  To work around this, you have to change your import like this:

```
from ui import Panel
```

and update the rest of your code to match.

This seems to be a limitation of the importing logic in the Pyjamas
source-code parser.  It's quite annoying, as leads to more verbose import
statements, and less clear code (is &quot;Panel&quot; something I defined, or part of
Pyjamas?  &quot;ui.Panel&quot; makes the origin of this class much clearer at a glance).

_Original issue: http://code.google.com/p/pyjamas/issues/detail?id=83 (September 09, 2008 08:12:21)_

**Comment by pyjsorg** [comment]
_From luke.lei...@gmail.com on September 15, 2008 20:07:25:_
ewestra: this is likely fixed in llpamie's branch.
the 0.12 pyjamas compiler has known limitations.

**Comment by pyjsorg** [comment]
_From luke.lei...@gmail.com on October 02, 2008 15:13:11:_
fixed this for the 0.3 release (pyjamas.sf.net)
